src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -10,16 +10,29 @@
     private ReadWriteLock lock = new ReentrantReadWriteLock();
 
     public void add(E element) {
-        // write your code here
+        lock.writeLock().lock();
```

Comment:
better create a local variable for ```lock.writeLock()``` because  you use it twice in this method (the same for other methods)